website/content/en/docs/building-operators/golang/advanced-topics.md

```diff
@@ -113,10 +113,196 @@ func init() {
 * After adding new import paths to your operator project, run `go mod vendor` if a `vendor/` directory is present in the root of your project directory to fulfill these dependencies.
 * Your 3rd party resource needs to be added before add the controller in `"Setup all Controllers"`.
 
-### Metrics
+### Monitoring and Observability
+This section aims to explain how to expose the first operator's custom metrics, [alerts] and [recording rules]. It focuses on the technical aspects, and demonstrates the implementation over the sample [memcached-operator]. 
 
-To learn about how metrics work in the Operator SDK read the [metrics section][metrics_doc] of the Kubebuilder documentation.
+For more information regarding operators' monitoring, please follow [observability-best-practices].  
 
+#### Prerequisites
+The following steps are required in order to inspect the operator's custom metrics, alerts and recording rules:
+- Deploy [kube-prometheus] on your cluster. This will set a Prometheus instance on the cluster, with which we will inspect the metrics, alerts and recording rules.
+- Make sure Prometheus has access to the operator's namespace, by setting the corresponding RBAC rules.
+  
+  Example: [prometheus_role.yaml] and [prometheus_role_binding.yaml]
+
+#### Publishing Custom Metrics
+If you wish to publish custom metrics for your operator, this can be easily achieved by using the global registry from `controller-runtime/pkg/metrics`.
```

Comment:
Do we also mention that doing this will also enable the generic controller-runtime metrics to be collected that are specified here: https://book.kubebuilder.io/reference/metrics-reference.html.

I'm not up-to-date on this though, would have to check if its enabled by default.

Reply:
These metrics are enabled by default